overlord/configstate/configmgr_test.go

```diff
@@ -0,0 +1,234 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configstate_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/corecfg"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord"
+	"github.com/snapcore/snapd/overlord/configstate"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+)
+
+type mockConf struct {
+	v map[string]interface{}
+}
+
+func (cfg *mockConf) Get(snapName, key string, result interface{}) error {
+	v := cfg.v[fmt.Sprintf("%s:%s", snapName, key)]
+
+	v1 := reflect.ValueOf(result)
+	v2 := reflect.Indirect(v1)
+	v2.Set(reflect.ValueOf(v))
+	return nil
+}
+
+type configmgrSuite struct {
+	o     *overlord.Overlord
+	state *state.State
+	mgr   *configstate.ConfigManager
+}
+
+var _ = Suite(&configmgrSuite{})
+
+func (s *configmgrSuite) SetUpTest(c *C) {
+	s.o = overlord.Mock()
+	s.state = s.o.State()
+
+	hookmgr, err := hookstate.Manager(s.state)
+	c.Assert(err, IsNil)
+	s.o.AddManager(hookmgr)
+
+	s.mgr, err = configstate.Manager(s.state, hookmgr)
+	c.Assert(err, IsNil)
+	s.o.AddManager(s.mgr)
```

Comment:
not related to this PR in itself, but I noticed that AddManager doesn't support configstate.Manager even now that is really a full manager

overlord.go needs fixing:

* AddManager needs a case for it
* this line in Overlord.New shouldn't be needed anymore:  
           o.configMgr = configMgr
*  we should add  an Overlord.ConfigManager